### PR TITLE
Update branch for debugpy 1.5.1 with last changes in main

### PR DESCRIPTION
**Pull request by paulacamargo25**
Updating branch that uses dubugpy 1.5.1 with last changes

### CODE REVIEW

File: .github/actions/build-vsix/action.yml [hunk 1]
```diff
@@ -30,23 +30,23 @@ runs:
           ${{ runner.os }}-pip-build-vsix-
 
     # For faster/better builds of sdists.
-    - name: Update pip, install pipx and install wheel
-      run: python -m pip install -U pip pipx wheel
+    - name: Update pip, install nox and install wheel
+      run: python -m pip install -U pip nox wheel
```
Comment (brettcannon) [inline .github/actions/build-vsix/action.yml:34]: Why is `wheel` installed?

And GitHub Actions includes pipx by default, so you should be able to skip installing both Nox and pipx and use `pipx run nox`.

File: .github/actions/lint/action.yml [hunk 2]
```diff
@@ -41,15 +41,15 @@ runs:
           ${{ runner.os }}-pip-lint-
 
     # For faster/better builds of sdists.
-    - name: Update pip, install pipx and install wheel
-      run: python -m pip install -U pip pipx wheel
+    - name: Update pip, install nox and install wheel
+      run: python -m pip install -U pip nox wheel
```
Comment (brettcannon) [inline .github/actions/lint/action.yml:45]: See my comments on the other action.

File: .github/actions/lint/action.yml [hunk 3]
```diff
@@ -30,7 +30,7 @@ runs:
     - name: Install Python
       uses: actions/setup-python@v4
       with:
-        python-version: '3.6'
+        python-version: '3.7'
```
Comment (brettcannon) [inline .github/actions/lint/action.yml:33]: If this is for https://pypi.org/project/debugpy/1.5.1/ then why the Python version bump?

File: noxfile.py
Comment (brettcannon) [inline noxfile.py:1]: Is this any different from what's in `main` and that I already reviewed previously?

File: package.json [hunk 4]
```diff
@@ -1,9 +1,12 @@
 {
     "name": "debugpy",
-    "displayName": "Debugpy",
+    "displayName": "Python Debugger",
     "description": "Python Debugger extension using `debugpy`.",
-    "version": "2023.1.0-dev",
+    "version": "2023.3.0-dev",
```
Comment (brettcannon) [inline package.json:5]: Why the version jump instead of a micro version bump?

**Review by brettcannon — CHANGES_REQUESTED** [review]

**Comment by brettcannon** [comment]
I can't follow what the purpose of this PR is. I see changes around `justMyCode` defaults, renames related to the extension name, and changing the version of Debugpy (while tweaking some actions). I feel like there are 3 or more PRs all wrapped into one here.

File: .github/actions/build-vsix/action.yml (hunk 1 above)
Comment (paulacamargo25) [reply .github/actions/build-vsix/action.yml:34]: I removed pipx because I got an error using python 3.6

File: noxfile.py
Comment (paulacamargo25) [reply noxfile.py:1]: no, im just updating the branch that uses debugpy 1.5.1 with the latest changes in main.

**Comment by brettcannon** [comment]
With https://github.com/microsoft/vscode-python/commit/3e7593712f3b58527bb38261992b845d358c079c having landed, is this PR still relevant? If so, should it be prioritized?

**Review by karthiknadig — APPROVED** [review]

File: src/extension/debugger/configuration/providers/pidAttach.ts [hunk 5]
```diff
@@ -20,7 +20,6 @@ export async function buildPidAttachConfiguration(
         type: DebuggerTypeName,
         request: 'attach',
         processId: '${command:pickProcess}',
-        justMyCode: true,
```
[the remark this answers is not in the document]
Comment (paulacamargo25) [reply src/extension/debugger/configuration/providers/pidAttach.ts:23]: yes

**Review by brettcannon — APPROVED** [review]

**Pull request merged**